jetty-util/src/main/java/org/eclipse/jetty/util/component/ContainerLifeCycle.java

```diff
@@ -106,18 +106,20 @@ protected void doStart() throws Exception
                     switch (b._managed)
                     {
                         case MANAGED:
-                            if (!l.isRunning())
+                            if (l.isStopped())
                                 start(l);
                             break;
 
                         case AUTO:
-                            if (l.isRunning())
-                                unmanage(b);
-                            else
+                            if (l.isStopped())
                             {
                                 manage(b);
                                 start(l);
                             }
+                            else
```

Comment:
Wouldn't this change make state FAILED trigger `unmanage(b)` now?
(Something that didn't happen before)

Reply:
I agree that `FAILED` should be treated like `STOPPED`